src/main/java/ch/wisv/events/EventsController.java

```diff
@@ -39,12 +40,51 @@ public String createEvent(Model model) {
         return "events/create";
     }
 
-    @RequestMapping(value = "/create", method = RequestMethod.POST)
+    @RequestMapping(value = "/save", method = RequestMethod.POST)
     @PreAuthorize("hasRole('ADMIN')")
-    public String createEvent(@ModelAttribute Event event, RedirectAttributes redirectAttributes) {
+    public String saveEvent(@ModelAttribute Event event, RedirectAttributes redirectAttributes) {
         event = eventRepository.save(event);
 
-        redirectAttributes.addFlashAttribute("message", "Event '" + event.getTitle() + "' created!");
+        redirectAttributes.addFlashAttribute("message", "Event '" + event.getTitle() + "' saved!");
         return "redirect:/events/";
     }
+
+    @RequestMapping(value = "/{id}", method = RequestMethod.GET)
+    public String getEvent(Model model, @PathVariable("id") long id, RedirectAttributes redirectAttributes) {
+        Event event = eventRepository.findOne(id);
+        if (event != null) {
+            model.addAttribute("event", event);
+            return "events/detail";
+        } else {
+            redirectAttributes.addFlashAttribute("message", "That event does not exist.");
+            return "redirect:/events/";
+        }
+    }
+
+    @RequestMapping(value = "/delete/{id}", method = RequestMethod.GET)
```

Comment:
It should definitely be a POST though. GET requests must never have side effects.